### PR TITLE
Navigate to the store if the search query is blank

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #2483.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- search for something using the header search box
- the results should appear
- click inside the search box and remove the text
- hit enter or click the magnifying glass and the store should appear

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7049/

File: jujugui/static/gui/src/app/components/header-search/test-header-search.js [hunk 1]
```diff
@@ -216,4 +216,35 @@ describe('HeaderSearch', function() {
       store: null
     });
   });
+
+  it('navigates to the store if the query is blank', function() {
+    appState.current = {
+      search: {text: ''},
+      activeComponent: 'store'
+    };
+    var renderer = jsTestUtils.shallowRender(
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/header-search/test-header-search.js]: const

File: jujugui/static/gui/src/app/components/header-search/test-header-search.js [hunk 2]
```diff
@@ -216,4 +216,35 @@ describe('HeaderSearch', function() {
...
+      <juju.components.HeaderSearch
+        appState={appState} />, true);
+    var instance = renderer.getMountedInstance();
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/header-search/test-header-search.js]: const

File: jujugui/static/gui/src/app/components/header-search/test-header-search.js [hunk 3]
```diff
@@ -216,4 +216,35 @@ describe('HeaderSearch', function() {
...
+    instance.refs = {searchInput: {focus: sinon.stub()}};
+    instance.state.active = true;
+    var output = renderer.getRenderOutput();
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/header-search/test-header-search.js]: const

**Comment by frankban** [comment]
QA is ok, but perhaps we should do the same with the stripped text, so that we display the store home even if the user just entered and empty string. But this is already an improvement, thanks!

**Comment by huwshimi** [comment]
@frankban good call, I'll update it.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7061/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7063/

**Comment by huwshimi** [comment]
Thanks @frankban @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**